projects/opendr_ws_2/src/opendr_simulation/opendr_simulation/human_model_generation_client.py

```diff
@@ -0,0 +1,108 @@
+        Send request to service assigned with the task to generate a human model from an image
+        :param img_rgb: The RGB image depicting a human
+        :type img_rgb: engine.data.Image
+        :param img_msk: The image, used as mask, for depicting a human's silhouette
+        :type img_msk: engine.data.Image
+        :param extract_pose: Defines whether to extract the pose of the depicted human or not
+        :type extract_pose: bool
+        :return: A tuple containing the generated human model and the extracted 3D pose
+        :rtype: tuple, (opendr.simulation.human_model_generation.utilities.model_3D.Model_3D, engine.target.Pose)
+        """
+        extract_pose_ros = Bool()
+        extract_pose_ros.data = extract_pose
+        self.req.img_rgb = self.bridge_cv.cv2_to_imgmsg(img_rgb, encoding="bgr8")
+        self.req.img_msk = self.bridge_cv.cv2_to_imgmsg(img_msk, encoding="bgr8")
+        self.req.extract_pose = extract_pose_ros
+        self.future = self.cli.call_async(self.req)
+        rclpy.spin_until_future_complete(self, self.future)
+        resp = self.future.result()
+        pose = self.bridge_ros.from_ros_pose_3D(resp.pose)
+        vertices, triangles = self.bridge_ros.from_ros_mesh(resp.mesh)
+        vertex_colors = self.bridge_ros.from_ros_colors(resp.vertex_colors)
+        human_model = Model_3D(vertices, triangles, vertex_colors)
+        return human_model, pose
+
+
+def main():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-srv_name", help="The name of the service",
+                        type=str, default="human_model_generation")
+    parser.add_argument("-img_rgb", help="Path for RGB image", type=str,
+                        default=os.path.join(os.environ['OPENDR_HOME'], 'projects/python/simulation/'
+                                                                        'human_model_generation/demos/'
+                                                                        'imgs_input/rgb/result_0004.jpg'))
+    parser.add_argument("-img_msk", help="Path for mask image", type=str,
+                        default=os.path.join(os.environ['OPENDR_HOME'], 'projects/python/simulation/'
+                                                                        'human_model_generation/demos/'
+                                                                        'imgs_input/msk/result_0004.jpg'))
+    parser.add_argument("-rot_angles", help="Yaw angles for rotating the generated model",
+                        nargs="+", default=['30', '120'])
+    parser.add_argument("-extract_pose", help="Whether to extract pose or not", action='store_true')
+    parser.add_argument("-plot_kps", help="Whether to plot the keypoints of the extracted pose",
+                        action='store_true')
+    parser.add_argument("-out_path", help="Path for outputting the renderings/models", type=str,
+                        default=os.path.join(os.environ['OPENDR_HOME'], 'projects/opendr_ws_2'))
```

Comment:
Argument names still look like they use single minus instead of double even though the comment and suggestion i made in earlier review is marked as resolved. I am re-adding the suggestion, reasoning can be found in earlier comment.


```suggestion
    parser.add_argument("--srv_name", help="The name of the service",
                        type=str, default="human_model_generation")
    parser.add_argument("--img_rgb", help="Path for RGB image", type=str,
                        default=os.path.join(os.environ['OPENDR_HOME'], 'projects/python/simulation/'
                                                                        'human_model_generation/demos/'
                                                                        'imgs_input/rgb/result_0004.jpg'))
    parser.add_argument("--img_msk", help="Path for mask image", type=str,
                        default=os.path.join(os.environ['OPENDR_HOME'], 'projects/python/simulation/'
                                                                        'human_model_generation/demos/'
                                                                        'imgs_input/msk/result_0004.jpg'))
    parser.add_argument("--rot_angles", help="Yaw angles for rotating the generated model",
                        nargs="+", default=['30', '120'])
    parser.add_argument("--extract_pose", help="Whether to extract pose or not", action='store_true')
    parser.add_argument("--plot_kps", help="Whether to plot the keypoints of the extracted pose",
                        action='store_true')
    parser.add_argument("--out_path", help="Path for outputting the renderings/models", type=str,
                        default=os.path.join(os.environ['OPENDR_HOME'], 'projects/opendr_ws_2'))
```

Reply:
Done